value_validation.go

```diff
@@ -0,0 +1,85 @@
+package structvalidator
+
+import (
+	"reflect"
+	"regexp"
+	"strings"
+)
+
+type ValueValidation struct {
+	LenMin int
+	LenMax int
+	ValMin int64
+	ValMax int64
+	Regexp *regexp.Regexp
+	Flags  int64
+}
+
+// values used with flags
+const (
+	_            = iota
+	ValMinNotNil = 1 << iota
+	ValMaxNotNil
+	Required
+	Email
+)
+
+func (v *ValueValidation) ValidateReflectValue(value reflect.Value) (ok bool, failureFlags int) {
+	minCanBeZero := false
+	maxCanBeZero := false
+	if v.Flags&ValMinNotNil > 0 {
+		minCanBeZero = true
+	}
+	if v.Flags&ValMaxNotNil > 0 {
+		maxCanBeZero = true
+	}
+
+	if v.Flags&Required > 0 {
+		if value.Type().Name() == "string" && value.String() == "" {
+			return false, FailEmpty
+		}
+		if strings.HasPrefix(value.Type().Name(), "int") && value.Int() == 0 && !minCanBeZero && !maxCanBeZero && v.ValMin == 0 && v.ValMax == 0 {
```

Comment:
Same here

```suggestion
		if (value.Kind() == reflect.Int || value.Kind() == reflect.Int64) && value.Int() == 0 && !minCanBeZero && !maxCanBeZero && v.ValMin == 0 && v.ValMax == 0 {
```

Reply:
This is a great suggestion.  And, I already have an `isInt` function that checks for int.  It's changed now, I'm resolving.